packages/zosfiles/src/methods/utilities/Utilities.ts

```diff
@@ -107,14 +108,36 @@ export class Utilities {
         const response = await Utilities.putUSSPayload(session, USSFileName, payload);
         const jsonObj = JSON.parse(response.toString());
         if (Object.prototype.hasOwnProperty.call(jsonObj, "stdout")) {
-            const stdout = JSON.parse(response.toString()).stdout[0];
+            const stdout = jsonObj.stdout[0];
             // Tests if binary tag set
-            return (stdout.indexOf("b ") >-1) ||
-            (stdout.indexOf("UTF-") >-1 ) || (stdout.indexOf("ISO8859-")>-1 ) || (stdout.indexOf("IBM-850") >-1 );
+            return (stdout.indexOf("b ") > -1) ||
+                (stdout.indexOf("UTF-") > -1) || (stdout.indexOf("ISO8859-") > -1) || (stdout.indexOf("IBM-850") > -1);
         }
         return false;
     }
 
+    /**
+     * Check the tags on an existing USS file and update binary and encoding
+     * properties on the options object.
+     * @param session z/OSMF connection info
+     * @param USSFileName Path to USS file
+     * @param options Options for downloading a USS file
+     */
+    public static async applyTaggedEncoding(session: AbstractSession, USSFileName: string, options: IOptions): Promise<void> {
+        const payload = { request: "chtag", action: "list" };
+        const response = await Utilities.putUSSPayload(session, USSFileName, payload);
+        const jsonObj = JSON.parse(response.toString());
+        if (Object.prototype.hasOwnProperty.call(jsonObj, "stdout")) {
+            const columns = (jsonObj.stdout[0] as string).trim().split(/\s+/);
+            // Tests if binary tag set
+            if (columns[0] === "b" || columns[1].startsWith("UTF-") || columns[1].startsWith("ISO8859-")) {
+                options.binary = true;
+            } else if (columns[1].startsWith("IBM-")) {
```

Comment:
How confident are we that the text trailing "IBM-" always represents a valid encoding value for the request?

Reply:
I was unable to find a table in IBM docs that provides a comprehensive list of allowed values for this field. However I believe this is a safe assumption to make, given that this seems to be a standard way of representing USS encoding. If you pass only a number to the chtag command (e.g., `chtag -tc 1147 <filename>`), the encoding field will be displayed as "IBM-1147" so the prefix gets added automatically. I suggest that if we discover more cases in the future which we don't yet support, a workaround is to use the `--binary` and `--encoding` options to manually set the encoding, and we can enhance our auto-detection method as these cases are discovered.

Reply:
👍

Reply:
I have since found a list with a number of encoding names [here](https://www.ibm.com/docs/en/zos/2.5.0?topic=utilities-code-set-converters-supplied), and some of them include a letter at the end (IBM-932C and IBM-943G). So it seems unsafe to be calling `parseInt` on the remainder of the string after `IBM-`, and I'll need to fix the parsing to make it more robust. Thanks for bringing this up, it prompted me to do some more research 🙂